### PR TITLE
Adding the Ecal Endcaps to the ML based online ECAL DQM

### PR DESCRIPTION
**Pull request by abhih1**
#### PR description:

This PR introduces the Ecal Endcaps into the autoencoder-based online ECAL DQM feature, which was implemented for EB in https://github.com/cms-sw/cmssw/pull/35990.
Separate Autoencoder (AE) models with ResNet architecture are trained for EE+ and EE-, apart from the model for EB, on certified good data (digi occupancy) from 2018 runs.

On giving an input occupancy map to the AE, the encoder part of the AE encodes and learns the features and the decoder reconstructs the data from the encoded latent space to match the input as closely as possible. The reconstruction loss is then calculated, which is a mean squared error (MSE) between the input and output images at a tower level. Thus given an anomalous tower, the AE which has learnt the features of the good data will have a hard time reconstructing it and give a higher loss on the anomaly than on the good towers. A quality threshold is then applied on this loss map which marks it as Good or Bad, which is then stored as an ML quality summary plot.
New correction factors are derived from 2022 collisions data to use in the pre-processing and inference, which follows the same steps as used for EB.

This PR thus introduces ML Quality summary plots for EE- and EE+, along with Loss Map and reconstructed occupancy maps from the AE.
It also introduces a trend plot to monitor the no. of bad towers flagged by the AE per lumisection in a run, as well as the map of these bad towers in an occupancy-like plot. This would be very helpful in monitoring per lumisection behaviour of bad towers/channels.

Please note that this PR should be tested along with the files added to https://github.com/cms-data/DQM-EcalMonitorClient/pull/3

#### PR validation:

The code was validated by running the online Ecal DQM configuration and the resultant plots were examined by uploading the output file to a DQM test gui.
The new plots are confirmed and look reasonable.

#### If this PR is a backport please specify the original PR and why you need to backport that PR. If this PR will be backported please specify to which release cycle the backport is meant for:
This is the master PR.
Backport to 13_0_X used in production is here: https://github.com/cms-sw/cmssw/pull/41195

### CODE REVIEW

**Comment by emanueleusai** [comment]
type ecal

**Comment by emanueleusai** [comment]
please test with https://github.com/cms-data/DQM-EcalMonitorClient/pull/3

**Comment by abhih1** [comment]
The errors seem to be unrelated to the changes in the PR.

**Comment by perrotta** [comment]
please test with https://github.com/cms-data/DQM-EcalMonitorClient/pull/3

**Comment by emanueleusai** [comment]
+1

**Comment by perrotta** [comment]
@abhih1 please solve conflict with the DQM/EcalMonitorClient/src/MLClient.cc now on master (it was updated for the new ONNXRuntime api by #41118)

**Comment by abhih1** [comment]
@perrotta Sure. Please see the new commit.

**Comment by perrotta** [comment]
please rebase on a recent IB, otherwise the conflict will not disappear in github

**Comment by abhih1** [comment]
> please rebase on a recent IB, otherwise the conflict will not disappear in github

@perrotta I have rebased. Hope this works now.

File: DQM/EcalMonitorClient/src/MLClient.cc [hunk 1]
```diff
@@ -12,6 +10,8 @@
 
 #include "DQM/EcalCommon/interface/MESetNonObject.h"
 
+#include <string>
```
Comment (perrotta) [inline DQM/EcalMonitorClient/src/MLClient.cc:13]: @abhih1 your [last commit](https://github.com/cms-sw/cmssw/pull/41175/commits/284e42604ba03db2cdc2d6b89d50e547176a845c) overwrites in a couple of places what was already implemented by https://github.com/cms-sw/cmssw/pull/41118 (for example here you are duplicating this `#include <string>`).

Can I suggest you to:
- remove the duplicate include
- add back the removed (empty) [line 296](https://github.com/cms-sw/cmssw/pull/41175/commits/284e42604ba03db2cdc2d6b89d50e547176a845c#diff-f6322d83172da74a703f6d7f55b6c0ee7950037ff3c3d56d89e2196d317d3595L296)
- squash all commits into one, to get rid of the latest, useless, updates

Comment (abhih1) [reply DQM/EcalMonitorClient/src/MLClient.cc]: @perrotta Sorry about that. Fixed and squashed all into one commit.

**Comment by perrotta** [comment]
please test

**Comment by emanueleusai** [comment]
+1

- resign

**Comment by perrotta** [comment]
+1

**Pull request merged**